test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -349,6 +355,34 @@ protected void afterIfFailed(List<Throwable> errors) {}
     /** called after a test is finished, but only if successful */
     protected void afterIfSuccessful() throws Exception {}
 
+    /**
+     * Marks a test suite or a test method that should run without security manager enabled.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE })
+    @Inherited
+    public @interface WithoutSecurityManager {
+    }
+
+    private static Closeable securityManagerRestorer;
```

Comment:
I'm not exactly sure where these before/after methods run, but we might want to make this field volatile, or use a lock to ensure visibility?

Reply:
I'm quite certain all test lifecycle methods execute within a single thread, and we don't perform any intra-JVM parallelization of test execution (parallel classes run in individual forked JVMs) so I don't think this is a concern.

Reply:
Yes that is right, all before*/after* methods are run in the same test thread, as if they were part of each test method execution. You can see things set in other tests by before/after do not concern themselves with threads.